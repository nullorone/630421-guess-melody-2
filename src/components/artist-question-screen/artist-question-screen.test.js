import React from 'react';
import renderer from 'react-test-renderer';
import ArtistQuestionScreen from "./artist-question-screen";

const Value = {
  EMPTY: 0,
  FULL: 1,
};

const EMPTY_ANSWER = {
  artist: ``,
  picture: {
    src: ``,
    alt: ``,
  },
};

describe(`Snapshot`, () => {
  const playButtonHandler = jest.fn();
  const answerHandler = jest.fn();
  const initialProps = {
    question: ``,
    answers: [EMPTY_ANSWER],
    onPlayButtonClick: playButtonHandler,
    onArtistAnswerClick: answerHandler,
    time: Value.EMPTY,
    mistakes: Value.EMPTY,
  };

  it(`Render`, () => {
    const tree = renderer
      .create(
          <ArtistQuestionScreen
            {...initialProps}
          />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
