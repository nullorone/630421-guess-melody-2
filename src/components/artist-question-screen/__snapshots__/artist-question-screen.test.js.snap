// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Render 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx="390"
        cy="390"
        r="370"
      />
    </svg>
    <div
      className="timer__value"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <span
        className="timer__mins"
      >
        00
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    />
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      
    </h2>
    <div
      className="game__track"
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          onClick={[MockFunction]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
      </div>
    </div>
    <form
      className="game__artist"
    >
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-1"
          name="answer"
          onClick={[Function]}
          type="radio"
          value="artist-1"
        />
        <label
          className="artist__name"
          htmlFor="answer-1"
        >
          <img
            alt=""
            className="artist__picture"
            src=""
          />
          
        </label>
      </div>
    </form>
  </section>
</section>
`;
